src/TodosContext.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useMemo, useState } from 'react';
+import { Todo, Status, Context } from './types';
+import { useLocalStorage } from './hooks/useLocalStorage';
+
+export const TodosContext = React.createContext({} as Context);
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [allTodos, setAllTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [todoEditId, setTodoEditId] = useState<number | null>(null);
+  const [status, setStatus] = useState(Status.All);
+
+  const visibleTodos = allTodos.filter(todo => {
+    switch (status) {
+      case Status.Active:
+        return !todo.completed;
+
+      case Status.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
```

Comment:
why do we need to iterate through todos with each render? add useMemo here
and also you can check like 

```
if (status === Status.All) return allTodos

return allTodos.filter(....)
```